src/finalizer/finalizer.nativeproj

```diff
@@ -0,0 +1,19 @@
+<!-- Platform needs to be set with TreatAsLocalProperty since it is a global property and cannot be overridden otherwise. -->
+<Project Sdk="Microsoft.DotNet.CMake.Sdk" TreatAsLocalProperty="Platform">
+
+  <PropertyGroup>
+    <Platform Condition="'$(Platform)' == '' Or $(Platform) == 'AnyCPU'">$(Architecture)</Platform>
+    <!-- We use forward slashes in the ArtifactsDir path because backslashes in the CMakeLists.txt are escape characters. -->
+    <ArtifactsDirForCMake>$([System.String]::Copy($(ArtifactsDir)).Replace('\','/'))</ArtifactsDirForCMake>
+    <CMakeLists>CMakeLists.txt</CMakeLists>
+  </PropertyGroup>
+
+  <!-- This will be passed through -D and referenced by the CMakeLists.txt to correctly resolve the platform specific libs. -->
+  <ItemGroup>
+    <CMakeDefines Include="Platform" Value="$(Platform)" />
+    <CMakeDefines Include="ArtifactsDir" Value="$(ArtifactsDirForCMake)" />
```

Comment:
We pass the directory explicitly instead of the `CMakeLists.txt` using relative `..` logic. This works well in this repo and the SDK repo since the artifacts output is the same directory structure.

Reply:
Would you mind trying out what happens if this directory path has spaces? it'll probably break the -D logic.
I haven't tried whether the repo builds today on a path with spaces though so we might already have issues.

Reply:
Looks like our build tools for this repo don't even work with spaces in the path:
![image](https://github.com/dotnet/installer/assets/17788297/51c9c2d7-2cc5-4e2b-a270-4e78cb45500b)

Let me fix that first.

Reply:
So, even after fixing the issue in `run-build.ps1`, the build still has other failures with spaces in the path. Certain targets fail and this CMake logic fails too. So, since multiple things are failing, I'm not going to change this logic right now. If we want builds to work with spaces, I'd rather fix it in the SDK repo after migration.

For future reference, this is what I did in the `try` in `run-build.ps1`:
```powershell
# See: https://paulcunningham.dev/powershell-invoke-expressions-spaces-in-path/
$buildScriptPath = "$RepoRoot\eng\common\build.ps1"
$buildScriptParameters = "-restore -build -nativeToolsOnMachine $Parameters $ExtraParameters"
Write-Host "Invoking expression: & '$buildScriptPath' $buildScriptParameters"
Invoke-Expression "& '$buildScriptPath' $buildScriptParameters"
```